packages/expo-sqlite/src/SQLite.ts

```diff
@@ -54,6 +93,26 @@ class SQLiteDatabase {
 
     return ExpoSQLite.deleteAsync(this._name);
   }
+
+  /**
+   * Creates a new transaction with Promise support.
+   * @param asyncCallback A `SQLTransactionAsyncCallback` function that can perform SQL statements in a transaction.
+   * @param readOnly true if all the SQL statements in the callback are read only.
+   */
+  async transactionAsync(
+    asyncCallback: SQLTransactionAsyncCallback,
+    readOnly: boolean = false
+  ): Promise<void> {
+    await this.execAsync([{ sql: 'BEGIN;', args: [] }], false);
+    try {
+      const transaction = new ExpoSQLTransactionAsync(this, readOnly);
```

Comment:
out of curiosity - how did you decide on wrapping this in an `ExpoSQLTransaction` class rather than inlining the `execAsync` here?

Reply:
the idea is to make `transactionAsync` like `transaction` but only for async mode. for `transaction`, it gives you [a transaction interface](https://github.com/expo/expo/blob/fb5dc6036cf10c1d288388690d482a8330b9be6d/packages/expo-sqlite/src/SQLite.types.ts#L63-L84) and that is actually a wrapped `WebSQLTransaction` as well.

i'm open to the design, feel free to comment if there's any thought.

Reply:
sounds reasonable, thanks for clarifying!